tests/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+enable_testing()
+find_package(GTest 1.10 REQUIRED)
+
+# Unit tests
+add_executable(cps-tests loader.cpp utils.cpp version.cpp)
+target_link_libraries(cps-tests PRIVATE libcps)
+target_link_libraries(cps-tests PRIVATE GTest::gtest GTest::gtest_main)
+
+include(GoogleTest)
+gtest_discover_tests(cps-tests)
+
+# Integration tests
+find_package(
+  Python 3.11
+  COMPONENTS Interpreter
+  REQUIRED)
+add_test(
+  NAME pkg-config-compatibility-tests
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+  COMMAND
+    ${CMAKE_COMMAND} -E env CPS_PATH=${CMAKE_CURRENT_SOURCE_DIR}/cases
```

Comment:
You can also set properties on tests to adjust the environment variables for the tests, but this implementation works too.

See:

* https://cmake.org/cmake/help/latest/command/set_tests_properties.html
* https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT_MODIFICATION.html